core/src/avm2/globals/string.rs

```diff
@@ -408,14 +408,43 @@ fn split<'gc>(
                     .into(),
             );
         }
+
+        let this = Value::from(this).coerce_to_string(activation)?;
         if delimiter
```

Comment:
This can be changed to:
```
if let Some(regexp_object) = delimeter.as_object().and_then(|o| o.as_regexp()) {
    ...
}
```
Then the `.unwrap()` inside the `if` can be avoided.

Reply:
I tried that but inside the if I need a mutable regexp, so I still need to evaluate `delimiter.as_object()`.

For this to work I'd need `o.as_regexp_mut()` inside `and_then(...)` but I found no way to return this from the closure.

Reply:
Actually nevermind, it seems I figured this one out!